lib/metadata.js

```diff
@@ -174,16 +174,21 @@ Metadata.prototype.updateOidcMetadata = function updateOidcMetadata(doc, next) {
 Metadata.prototype.generateOidcPEM = function generateOidcPEM(kid) {
   const keys = this && this.oidc && Array.isArray(this.oidc.keys) ? this.oidc.keys : null;
   let pubKey = null;
+  let foundKey = false;
 
-  if (!(kid && keys)) {
-    return null;
+  if (!kid) {
+    throw new Error('kid is missing');
+  }
+
+  if (!keys) {
+    throw new Error('keys is missing');
   }
 
   keys.some((key) => {
     log.info('working on key:', key);
 
     // are we working on the right key?
-    if (!key.kid === kid) {
+    if (key.kid !== kid) {
```

Comment:
Yes, but I haven't got ship it for that PR yet. It would be great if that PR can be reviewed soon since it's important.

https://github.com/AzureAD/passport-azure-ad/pull/139